catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/query/GeoNamesQueryLuceneIndex.java

```diff
@@ -285,4 +276,63 @@ query searching for any of the terms in the query string (the OR query). We take
             throw new GeoEntryQueryException("Error reading the index", e);
         }
     }
+
+    protected String doGetCountryCode(Shape shape, int radiusInKm, Directory directory)
+            throws GeoEntryQueryException {
+        notNull(shape,
+                "GeoNamesQueryLuceneIndex.doGetCountryCode(): argument 'shape' may not be null.");
+
+        if (radiusInKm <= 0) {
+            throw new IllegalArgumentException(
+                    "GeoNamesQueryLuceneIndex.doGetCountryCode(): radiusInKm must be positive.");
+        }
+
+        if (directory == null) {
+            return null;
+        }
+
+        try (final IndexReader indexReader = createIndexReader(directory)) {
+
+            final IndexSearcher indexSearcher = createIndexSearcher(indexReader);
+
+            final Point center = shape.getCenter();
+
+            final Filter filter = createSpatialFilter(center, radiusInKm);
+
+            final BooleanQuery booleanQuery = new BooleanQuery();
+            booleanQuery.add(filter, BooleanClause.Occur.FILTER);
+
+            final TopDocs topDocs = indexSearcher.search(booleanQuery, 1, SORT);
+
+            String countryCode = null;
+
+            if (topDocs.totalHits > 0) {
+                for (ScoreDoc scoreDoc : topDocs.scoreDocs) {
```

Comment:
Why not just take the first result?
`countryCode = indexSearcher.doc(topDocs.scoreDocs[0].doc).get(COUNTRY_CODE_FIELD);`